src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -251,6 +251,21 @@ public boolean hasTag(String tag) {
 
 							boolean success = element.init(parseResult.exprs, patternIndex, getParser().getHasDelayBefore(), parseResult);
 							if (success) {
+								for (Expression<?> expr : parseResult.exprs) {
+									if (!(expr instanceof UnparsedLiteral unparsedLiteral))
+										continue;
+									if (unparsedLiteral.wasReparsed() || unparsedLiteral.wasConverted())
+										continue;
+									if (!Classes.patternHasMultipleInfos(unparsedLiteral.getData()))
+										continue;
+									String unparsedPattern = unparsedLiteral.getData();
+									List<ClassInfo<?>> patternInfos = Classes.getPatternInfos(unparsedPattern);
+									assert patternInfos != null;
+									String infoCodeName = patternInfos.get(0).getName().getSingular();
+									Skript.warning("'" +  unparsedPattern + "' has multiple types. Consider specifying which type to use: '"
+										+ unparsedPattern + " (" + infoCodeName + ")'");
+									break;
+								}
```

Comment:
we shouldn't be bloating this method further. Use helper methods, could even make it a unparsedliteral method.